rewrite-core/src/main/java/org/openrewrite/semver/LatestRelease.java

```diff
@@ -72,14 +71,21 @@ static String normalizeVersion(String version) {
 
     static long countVersionParts(String version) {
         long count = 0;
-        Scanner scanner = new Scanner(version);
-        scanner.useDelimiter("[.\\-$]");
-        while (scanner.hasNext()) {
-            String part = scanner.next();
-            if (part.isEmpty() || !Character.isDigit(part.charAt(0))) {
-                break;
+        int len = version.length();
+        int lastSepIdx = -1;
+        for (int i = 0; i < len; i++) {
+            char c = version.charAt(i);
+            if (c == '.' || c == '-' || c == '$') {
```

Comment:
This is slightly different from the delimiter regex which originally also included "\". But maybe that doesn't matter? Haven't ever seen a backslash in a version number

Reply:
I think the backslash in `[.\\-$]` is so that the regex doesn't interpret it as a range, as it is in square brackets.